src/tests/Microsoft.PowerFx.Core.Tests.Shared/ExpressionTestCases/JSON.txt

```diff
@@ -133,3 +133,60 @@ Error({Kind:ErrorKind.Div0})
 // Flattening nested tables
 >> JSON([[1,2,3],[4,5],[6]], JSONFormat.FlattenValueTables)
 "[[1,2,3],[4,5],[6]]"
+
+>> JSON(ParseJSON("{}"))
+Errors: Error 5-20: The JSON function cannot serialize objects of type 'UntypedObject'.
+
+>> JSON(ParseJSON("[]"))
+Errors: Error 5-20: The JSON function cannot serialize objects of type 'UntypedObject'.
+
+>> JSON(ParseJSON("{}"), JSONFormat.IgnoreUnsupportedTypes)
+"{}"
+
+>> JSON(ParseJSON("[]"), JSONFormat.IgnoreUnsupportedTypes)
+"[]"
+
+>> JSON(ParseJSON("1"), JSONFormat.IgnoreUnsupportedTypes)
+"1"
+
+>> JSON(ParseJSON("1.77"), JSONFormat.IgnoreUnsupportedTypes)
+"1.77"
+
+>> JSON(ParseJSON("-871"), JSONFormat.IgnoreUnsupportedTypes)
+"-871"
+
+>> JSON(ParseJSON("""John"""), JSONFormat.IgnoreUnsupportedTypes)
+"""John"""
+
+>> JSON(ParseJSON("true"), JSONFormat.IgnoreUnsupportedTypes)
+"true"
+
+>> JSON(ParseJSON("false"), JSONFormat.IgnoreUnsupportedTypes)
+"false"
+
+>> JSON(ParseJSON("{""a"": 1}"), JSONFormat.IgnoreUnsupportedTypes)
```

Comment:
>ParseJSON("{""a"": 1}") [](http://example.com/codeflow?start=8&length=23)

This test already shows it briefly, but we should have one case where it is more explicit that whitespaces are not round-tripped, something like `JSON(ParseJSON("   {\n  ""a"" : \t1\r}\r\n"))` ==> `"{""a"":1}"`